src/arrayMethodSort.js

```diff
@@ -4,8 +4,38 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const initialCompareFunction = (a, b) => {
+      if (String(a) < String(b)) {
+        return -1;
+      }
+
+      return 1;
+    };
+
+    const compare = compareFunction ?? initialCompareFunction;
+
+    let isSwapped = false;
+
+    do {
+      isSwapped = false;
+
+      for (let idx = 0; idx < this.length; idx++) {
```

Comment:
The loop should iterate until `this.length - 1` to avoid comparing an element with `undefined`. Accessing `this[idx + 1]` when `idx` is `this.length - 1` will result in `undefined`, which can lead to incorrect sorting behavior.